cmd/cfssl/cfssl_test.go

```diff
@@ -1 +1,350 @@
+
+// Var which holds the CA Request which is used to create
+// our CA Certificate to start CFSSL server
+var (
+	keyRequest = csr.BasicKeyRequest{
+		A: "rsa",
+		S: 2048,
+	}
+	CAConfig = csr.CAConfig{
+		PathLength: 1,
+		Expiry:     "1/1/2016",
+	}
+	CARequest = csr.CertificateRequest{
+		CN: "example.com",
+		Names: []csr.Name{
+			{
+				C:  "US",
+				ST: "California",
+				L:  "San Francisco",
+				O:  "Internet Widgets, LLC",
+				OU: "Certificate Authority",
+			},
+		},
+		Hosts:      []string{"ca.example.com"},
+		KeyRequest: &keyRequest,
+		CA:         &CAConfig,
+	}
+)
+
+/* We will Unmarshal API and CLI outputs into the following structs */
+type Response struct {
+	Success bool `json:"success"`
+}
+
+type CLIBundleResponse struct {
+	Bundle string `json:"bundle"`
+	CRT    string `json:"crt"`
+	Root   string `json:"root"`
+}
+
+type APIBundleResponse struct {
+	Response
+	Result struct {
+		CLIBundleResponse
+	} `json:"result"`
+}
+
+type CLIScanResponse struct {
+	Connectivity struct {
+		DNSLookup struct {
+			Output []string `json:"output"`
+		} `json:"DNSLookup"`
+	} `json:"Connectivity"`
+	PKI struct {
+		ChainExpiration struct {
+			Output string `json:"output"`
+		} `json:"ChainExpiration"`
+	} `json:"PKI"`
+	TLSHandshake struct {
+		CertsByCiphers struct {
+			Output struct {
+				TLS string `json:"TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA"`
+			} `json:"output"`
+		} `json:"CertsByCiphers"`
+	} `json:"TLSHandshake"`
+}
+
+type APIScanResponse struct {
+	Response
+	Result struct {
+		CLIScanResponse
+	} `json:"result"`
+}
+
+type CLICertInfoResponse struct {
+	Pem        string `json:"pem"`
+	CommonName string `json:"common_name"`
+	Sigalg     string `json:"sigalg"`
+}
+
+type APICertInfoResponse struct {
+	Response
+	Result struct {
+		CLICertInfoResponse
+	} `json:"result"`
+}
+
+type CLISignResponse struct {
+	Response
+	Certificate string `json:"certificate"`
+}
+
+type APISignResponse struct {
+	Response
+	Result struct {
+		CLISignResponse
+	} `json:"result"`
+}
+
+/* Start up the CFSSL local server. */
+func setup() {
+	// Set up a test server using our CA certificate and key.
+	CACert, CAKey, err := testsuite.CreateSelfSignedCert(CARequest)
+	if err != nil {
+		panic(err)
+	}
+	// Set up a test server using our CA certificate and key.
+	serverData := testsuite.CFSSLServerData{CA: CACert, CAKey: CAKey}
+	s, err := testsuite.StartCFSSLServer(addressToTest, portToTest, serverData)
```

Comment:
```:=``` apparently will both assign and initialize a variable in Go. We'd want to assign an existing variable ```server``` and initialize/assign nonexisting variable ```err``` at the same time. It looks like setting ```server``` equal to the initialized return of ```testsuite.StartCFSSLServer``` is the best option in this case.